### PR TITLE
Tagging - Add custom stencil text tags

### PR DESCRIPTION
**Pull request by PabstMirror**
Add stencil text tags
Using https://fonts.google.com/specimen/Saira+Stencil+One (Open Font License)

todo:
- [x] not sure cause of `Warning Message: Cannot load mipmap z\ace\addons\fonts\sairastencilone\sairastencilone46-01.paa`

combined with vehicle "clan" tagging:

adds 3den attribute for this

### CODE REVIEW

**Comment by PabstMirror** [comment]
23Cdrrn
```
Warning Message: Cannot load mipmap z\ace\addons\fonts\pixelsplitterbold\pixelsplitterbold46-01.paa
Extreme texture size (1x1)
Texture init failed.
```

But doing a dummy call to `"Test" getTextWidth ["ACE_Stencil", 0.3];` at preStart fixes
I'm pretty sure this is just an arma bug because it looks like all font textures should be preloaded

**Comment by jonpas** [comment]
How does this work in combination with `novehicleclanlogo` being enabled?

**Review by veteran29 — APPROVED** [review]

File: addons/tagging/CfgEden.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        class AttributeCategories {
+            class ace_attributes {
+                class Attributes {
+                    class GVAR(stencilVehicle) {
+                        property = QGVAR(stencilVehicle);
+                        control = "Edit";
+                        displayName = CSTRING(stencilVehicle);
+                        tooltip = CSTRING(stencilVehicle_tooltip);
+                        expression = QUOTE( [ARR_2(_this,_value)] call FUNC(stencilVehicle); );
+                        condition = "objectVehicle";
```
Comment (jonpas) [inline addons/tagging/CfgEden.hpp:12]:
```suggestion
                        condition = "objectVehicle";
```

File: addons/tagging/functions/fnc_addStencilTag.sqf [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#include "script_component.hpp"
+/*
+ * Author: PabstMirror
+ * Adds custom text tag. Has to be executed on one machine only.
+ *
+ * Arguments:
+ * 0: Display Text <STRING>
+ * 1: Text Size <NUMBER> (default: 0.3)
+ * 2: Required Item <STRING> (default: ACE_SpraypaintBlack)
```
Comment (jonpas) [inline addons/tagging/functions/fnc_addStencilTag.sqf:9]:
```suggestion
 * 2: Required Item <STRING> (default: "ACE_SpraypaintBlack")
```

File: addons/tagging/functions/fnc_addStencilTag.sqf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * 3: Text Color (in HEX 6 or 8) <STRING> (default: based on spray item)
+ * 4: Background Color (in HEX 6 or 8) <STRING> (default: clear)
```
Comment (jonpas) [inline addons/tagging/functions/fnc_addStencilTag.sqf:11]:
```suggestion
 * 4: Background Color (in HEX 6 or 8) <STRING> (default: "clear")
```

File: addons/tagging/functions/fnc_generateStencilTexture.sqf [hunk 4]
```diff
@@ -0,0 +1,55 @@
+#include "script_component.hpp"
+/*
+ * Author: PabstMirror
+ * Generate a "Text to Texture"
+ *
+ * Arguments:
+ * 0: Display Text <STRING>
+ * 1: Text Size <NUMBER> (default: 0.3)
+ * 2: Text Color (in HEX 6 or 8) <STRING> (default: black)
+ * 3: Background Color (in HEX 6 or 8) <STRING> (default: clear)
```
Comment (jonpas) [inline addons/tagging/functions/fnc_generateStencilTexture.sqf:10]:
```suggestion
 * 2: Text Color (in HEX 6 or 8) <STRING> (default: "black")
 * 3: Background Color (in HEX 6 or 8) <STRING> (default: "clear")
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/tagging/functions/fnc_addStencilTag.sqf (hunk 3 above)
Comment (PabstMirror) [reply addons/tagging/functions/fnc_addStencilTag.sqf:11]:
```suggestion
 * 4: Background Color (in HEX 6 or 8) <STRING> (default: "00000000" - transparent)
```

File: addons/tagging/functions/fnc_generateStencilTexture.sqf (hunk 4 above)
Comment (PabstMirror) [reply addons/tagging/functions/fnc_generateStencilTexture.sqf:10]:
```suggestion
 * 2: Text Color (in HEX 6 or 8) <STRING> (default: black)
 * 3: Background Color (in HEX 6 or 8) <STRING> (default: "00000000" - transparent)
```

File: addons/tagging/functions/fnc_generateStencilTexture.sqf [hunk 5]
```diff
@@ -0,0 +1,55 @@
+#include "script_component.hpp"
+/*
+ * Author: PabstMirror
+ * Generate a "Text to Texture"
+ *
+ * Arguments:
+ * 0: Display Text <STRING>
+ * 1: Text Size <NUMBER> (default: 0.3)
+ * 2: Text Color (in HEX 6 or 8) <STRING> (default: black)
```
Comment (PabstMirror) [inline addons/tagging/functions/fnc_generateStencilTexture.sqf:9]:
```suggestion
 * 2: Text Color (in HEX 6 or 8) <STRING> (default: "000000" - black)
```

**Comment by jonpas** [comment]
> How does this work in combination with `novehicleclanlogo` being enabled?

Would still like to know that.

**Comment by LinkIsGrim** [comment]
> > How does this work in combination with `novehicleclanlogo` being enabled?
> 
> Would still like to know that.

Stencil is removed on mission start.

**Comment by LinkIsGrim** [comment]
Player interaction for this would be nice?

**Review by jonpas — APPROVED** [review]

**Pull request merged**